### PR TITLE
[bitnami/valkey] Update all the references from to master/slave to primary/replica

### PR DESCRIPTION
**Pull request by dgomezleon**
### Description of the change

This major updates all the references from `master/slave` to `primary/replica` to follow the upstream project strategy. Environment variables previously prefixed as `VALKEY_MASTER` or `VALKEY_SENTINEL_MASTER` use `VALKEY_PRIMARY` and `VALKEY_SENTINEL_PRIMARY` now.

### Benefits

Use upstream terminology

### Possible drawbacks

Setting need to be updated to match `primary` instead of `master`

### Applicable issues

- fixes #

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/valkey/README.md [hunk 1]
```diff
@@ -1011,6 +1011,12 @@ helm install my-release -f values.yaml oci://REGISTRY_NAME/REPOSITORY_NAME/valke
 
 Find more information about how to deal with common errors related to Bitnami's Helm charts in [this troubleshooting guide](https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues).
 
+## Upgrading
+
+### To 2.0.0
+
+This major updates all the references from `master/slave` to `primary/replica` to follow the upstream project strategy. Environment variables previously prefixed as `VALKEY_MASTER` or `VALKEY_SENTINEL_MASTER` use `VALKEY_PRIMARY` and `VALKEY_SENTINEL_PRIMARY` now.
```
Comment (juan131) [inline bitnami/valkey/README.md:1018]: I'm missing upgradability instructions. How can I upgrade my existing Valkey workloads from `1.y.z` to `2.y.z`?

File: bitnami/valkey/templates/primary/pvc.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{{- /*
+Copyright Broadcom, Inc. All Rights Reserved.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+{{- if and (eq .Values.architecture "standalone") (eq .Values.primary.kind "Deployment") (.Values.primary.persistence.enabled) (not .Values.primary.persistence.existingClaim) }}
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: {{ printf "valkey-data-%s-primary" (include "common.names.fullname" .) }}
```
Comment (juan131) [inline bitnami/valkey/templates/primary/pvc.yaml:10]: This name can potentially be very long, hence we should ensure it meets K8s restrictions:
```suggestion
  name: {{ printf "valkey-data-%s-primary" (include "common.names.fullname" .) | trunc 63 | trimSuffix "-" }}
```

File: bitnami/valkey/templates/scripts-configmap.yaml [hunk 3]
```diff
@@ -74,39 +74,39 @@ data:
...
+        VALKEY_SERVICE="$VALKEY_EXTERNAL_PRIMARY_HOST"
     else
         VALKEY_SERVICE="{{ template "common.names.fullname" . }}.{{ include "common.names.namespace" . }}.svc.{{ .Values.clusterDomain }}"
     fi
 
     SENTINEL_SERVICE_PORT=$(get_port "{{ include "common.names.fullname" . }}" "SENTINEL")
     validate_quorum() {
         if is_boolean_yes "$VALKEY_TLS_ENABLED"; then
-            quorum_info_command="{{- if and .Values.auth.enabled .Values.auth.sentinel }}REDISCLI_AUTH="\$VALKEY_PASSWORD" {{ end }}valkey-cli -h $VALKEY_SERVICE -p $SENTINEL_SERVICE_PORT --tls --cert ${VALKEY_TLS_CERT_FILE} --key ${VALKEY_TLS_KEY_FILE} --cacert ${VALKEY_TLS_CA_FILE} sentinel master {{ .Values.sentinel.masterSet }}"
+            quorum_info_command="{{- if and .Values.auth.enabled .Values.auth.sentinel }}REDISCLI_AUTH="\$VALKEY_PASSWORD" {{ end }}valkey-cli -h $VALKEY_SERVICE -p $SENTINEL_SERVICE_PORT --tls --cert ${VALKEY_TLS_CERT_FILE} --key ${VALKEY_TLS_KEY_FILE} --cacert ${VALKEY_TLS_CA_FILE} sentinel primary {{ .Values.sentinel.primarySet }}"
```
Comment (juan131) [inline bitnami/valkey/templates/scripts-configmap.yaml:86]: Can we confirm the "sentinel primary" command exists in the `valkey-cli`?

File: bitnami/valkey/templates/scripts-configmap.yaml [hunk 4]
```diff
@@ -74,39 +74,39 @@ data:
...
 
         info "about to run the command: $failover_command"
         eval $failover_command
     }
 
-    get_sentinel_master_info() {
+    get_sentinel_primary_info() {
         if is_boolean_yes "$VALKEY_TLS_ENABLED"; then
-            sentinel_info_command="{{- if and .Values.auth.enabled .Values.auth.sentinel }}REDISCLI_AUTH="\$VALKEY_PASSWORD" {{ end }}timeout {{ .Values.sentinel.getMasterTimeout }} valkey-cli -h $VALKEY_SERVICE -p $SENTINEL_SERVICE_PORT --tls --cert ${VALKEY_TLS_CERT_FILE} --key ${VALKEY_TLS_KEY_FILE} --cacert ${VALKEY_TLS_CA_FILE} sentinel get-master-addr-by-name {{ .Values.sentinel.masterSet }}"
+            sentinel_info_command="{{- if and .Values.auth.enabled .Values.auth.sentinel }}REDISCLI_AUTH="\$VALKEY_PASSWORD" {{ end }}timeout {{ .Values.sentinel.getPrimaryTimeout }} valkey-cli -h $VALKEY_SERVICE -p $SENTINEL_SERVICE_PORT --tls --cert ${VALKEY_TLS_CERT_FILE} --key ${VALKEY_TLS_KEY_FILE} --cacert ${VALKEY_TLS_CA_FILE} sentinel get-primary-addr-by-name {{ .Values.sentinel.primarySet }}"
```
Comment (juan131) [inline bitnami/valkey/templates/scripts-configmap.yaml:107]: Same here, can we confirm the "sentinel get-primary-addr-by-name" command exists in the `valkey-cli`?

**Review by juan131 — CHANGES_REQUESTED** [review]

File: bitnami/valkey/templates/scripts-configmap.yaml (hunk 3 above)
Comment (dgomezleon) [reply bitnami/valkey/templates/scripts-configmap.yaml:86]: [Link](https://github.com/valkey-io/valkey/releases/tag/8.0.0-rc2#:~:text=Add%204%20sentinel%20command%20GET%2DPRIMARY%2DADDR%2DBY%2DNAME%2C%20PRIMARY%2C%20PRIMARIES%2C%20and%0AIS%2DPRIMARY%2DDOWN%2DBY%2DADDR%20to%20allow%20clients%20to%20use%20inclusive%20language.)

File: bitnami/valkey/templates/scripts-configmap.yaml (hunk 4 above)
Comment (dgomezleon) [reply bitnami/valkey/templates/scripts-configmap.yaml:107]: [Link](https://github.com/valkey-io/valkey/releases/tag/8.0.0-rc2#:~:text=Add%204%20sentinel%20command%20GET%2DPRIMARY%2DADDR%2DBY%2DNAME%2C%20PRIMARY%2C%20PRIMARIES%2C%20and%0AIS%2DPRIMARY%2DDOWN%2DBY%2DADDR%20to%20allow%20clients%20to%20use%20inclusive%20language.)

File: bitnami/valkey/templates/scripts-configmap.yaml (hunk 3 above)
Comment (dgomezleon) [reply bitnami/valkey/templates/scripts-configmap.yaml:86]: Also tested:

```
$ REDISCLI_AUTH="$VALKEY_PASSWORD" valkey-cli -h valkey -p 26379 sentinel primary myprimary
 1) "name"
 2) "myprimary"
 3) "ip"
 4) "valkey-node-0.valkey-headless.test.svc.cluster.local"
 5) "port"
 6) "6380"
...
```

File: bitnami/valkey/templates/scripts-configmap.yaml (hunk 4 above)
Comment (dgomezleon) [reply bitnami/valkey/templates/scripts-configmap.yaml:107]: Also tested:

```
$ REDISCLI_AUTH="$VALKEY_PASSWORD" valkey-cli -h valkey -p 26379 sentinel primary myprimary
 1) "name"
 2) "myprimary"
 3) "ip"
 4) "valkey-node-0.valkey-headless.test.svc.cluster.local"
 5) "port"
 6) "6380"
```

File: bitnami/valkey/README.md [hunk 5]
```diff
@@ -107,56 +107,56 @@ See the [official ExternalDNS documentation](https://github.com/kubernetes-sigs/
...
+This command will return the address of the current primary, which can be accessed from inside the cluster.
 
-In case the current master crashes, the Sentinel containers will elect a new master node.
+In case the current primary crashes, the Sentinel containers will elect a new primary node.
 
-`master.replicaCount` greater than `1` is not designed for use when `sentinel.enabled=true`.
+`primary.replicaCount` greater than `1` is not designed for use when `sentinel.enabled=true`.
 
-### Multiple masters (experimental)
+### Multiple primarys (experimental)
```
Comment (juan131) [inline bitnami/valkey/README.md:147]: Change references to "primarys" to "primary nodes"
```suggestion
### Multiple primary nodes (experimental)
```

File: bitnami/valkey/README.md [hunk 6]
```diff
@@ -1011,6 +1011,30 @@ helm install my-release -f values.yaml oci://REGISTRY_NAME/REPOSITORY_NAME/valke
...
+- Create a backup of the database, and restore it on the new release as explained in the [Backup and restore](#backup-and-restore) section.
+- Reuse the PVC used to hold the master data on your previous release. To do so, use the `primary.persistence.existingClaim` parameter. The following example assumes that the release name is `valkey`:
+
+```console
+helm install valkey oci://REGISTRY_NAME/REPOSITORY_NAME/valkey --set auth.password=[PASSWORD] --set primary.persistence.existingClaim=[EXISTING_PVC]
+```
+
+> Note: You need to substitute the placeholders `REGISTRY_NAME` and `REPOSITORY_NAME` with a reference to your Helm chart registry and repository. For example, in the case of Bitnami, you need to use `REGISTRY_NAME=registry-1.docker.io` and `REPOSITORY_NAME=bitnamicharts`.
+
+| Note: you need to substitute the placeholder *[EXISTING_PVC]* with the name of the PVC used on your previous release, and *[PASSWORD]* with the password used in your previous release.
```
Comment (juan131) [inline bitnami/valkey/README.md:1036]: Nice

**Review by juan131 — DISMISSED** [review]
LGTM!

**Review by juan131 — APPROVED** [review]

**Comment by vdmkenny** [comment]
What is the migration strategy here? This will destroy our current PVCs and data within. We have dozens of instances of the old chart with master, and we don't want to delete these existing PVCs.

**Pull request merged**